kedro/runner/task.py

```diff
@@ -189,67 +187,85 @@ def _run_node_sequential(
             )
         return node
 
-    def _run_node_async(
+    async def _run_node_async(
         self,
         node: Node,
         catalog: CatalogProtocol,
         hook_manager: PluginManager,  # type: ignore[arg-type]
         session_id: str | None = None,
     ) -> Node:
-        with ThreadPoolExecutor() as pool:
-            inputs: dict[str, Future] = {}
+        input_tasks: dict[str, asyncio.Task] = {
+            name: asyncio.create_task(
+                self._async_dataset_load(name, node, catalog, hook_manager)
+            )
+            for name in node.inputs
+        }
 
-            for name in node.inputs:
-                inputs[name] = pool.submit(
-                    self._synchronous_dataset_load, name, node, catalog, hook_manager
-                )
+        # Gather results as they complete
+        inputs = {}
+        for name, input_task in input_tasks.items():
+            try:
+                inputs[name] = await input_task
+            except Exception as exception:  # pragma: no cover
+                raise exception
 
-            wait(inputs.values(), return_when=ALL_COMPLETED)
-            inputs = {key: value.result() for key, value in inputs.items()}
-            is_async = True
-            additional_inputs = self._collect_inputs_from_hook(
-                node, catalog, inputs, is_async, hook_manager, session_id=session_id
-            )
-            inputs.update(additional_inputs)
+        is_async = True
+        additional_inputs = self._collect_inputs_from_hook(
+            node, catalog, inputs.copy(), is_async, hook_manager, session_id=session_id
+        )
+        inputs.update(additional_inputs)
 
-            outputs = self._call_node_run(
-                node, catalog, inputs, is_async, hook_manager, session_id=session_id
+        outputs = self._call_node_run(
+            node, catalog, inputs, is_async, hook_manager, session_id=session_id
+        )
+
+        output_tasks = {
+            name: asyncio.create_task(
+                self._async_dataset_save(name, data, node, catalog, hook_manager)
             )
+            for name, data in outputs.items()
+        }
+
+        for output_task in asyncio.as_completed(output_tasks.values()):
+            try:
+                await output_task
+            except Exception as exception:
```

Comment:
Same question here for the exception. Or at least could we make the message below more specific to help with debugging?